src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -123,6 +125,20 @@ namespace dpdk {
         args.push_back(string2vector("--no-huge"));
       }
 
+      std::list<std::string> port_list;
+      get_str_list(cct->_conf.get_val<std::string>("ms_dpdk_devs_allowlist"), " ", port_list);
+      for (auto &i : port_list) {
+        size_t equal = i.find('=');
+        if (equal != std::string::npos) {
+           const std::string &key = i.substr(0,equal);
```

Comment:
i am afraid that this is wrong. we cannot keep a reference of a rvalue. so `key` would have been destructed when it is passed to `string2vector()`.

BTW, i created #44353 which allows developer to iterate through a str which includes key-value pairs using `for_each_pair()`.

before #44353 is reviewed and merged. i'd suggest just use
```c++
if (equal != i.npos) {
  const std::string key = i.substr(0, equal);
  // ...
}
```

Reply:
> i am afraid that this is wrong. we cannot keep a reference of a rvalue. so `key` would have been destructed when it is passed to `string2vector()`.
> 
> BTW, i created #44353 which allows developer to iterate through a str which includes key-value pairs using `for_each_pair()`.
> 
> before #44353 is reviewed and merged. i'd suggest just use
> 
> ```c++
> if (equal != i.npos) {
>   const std::string key = i.substr(0, equal);
>   // ...
> }
> ```

Okay, I will update after for_each_pair merge

Reply:
@tchaikov The latest API has been used for modification. Please review the modification.

Reply:
@fengchunsong commented.